### PR TITLE
overlay: Diffsize: avoid naive diff

### PR DESCRIPTION
**Pull request by vrothberg**
Computing the diff size for the rootful overlay storage driver used the naive diff.  The reasoning was that we have made use of rootful copy-up. The downside is a mind-blowing performance penalty in comparison to the rootless case.

Hence, drop the naive diff and only compute the size of the directory which is absolutely sufficient for the motivating use case of podman-system-df.  This drops the execution of system-df from 19.3s to 0.11s listing 5 containers and 1 image.

Fixes: github.com/containers/podman/issues/19467

### CODE REVIEW

**Comment by vrothberg** [comment]
@giuseppe @rhatdan PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
@giuseppe @nalind I merged, but would like to have you comment on this

**Comment by giuseppe** [comment]
that is the patch I've suggested in https://github.com/containers/podman/issues/19467#issuecomment-1677009528 so I am fine with this approach

File: drivers/overlay/overlay.go [hunk 1]
```diff
@@ -2162,10 +2162,6 @@ func (d *Driver) getLowerDiffPaths(id string) ([]string, error) {
 // and its parent and returns the size in bytes of the changes
 // relative to its base filesystem directory.
 func (d *Driver) DiffSize(id string, idMappings *idtools.IDMappings, parent string, parentMappings *idtools.IDMappings, mountLabel string) (size int64, err error) {
-	if d.options.mountProgram == "" && (d.useNaiveDiff() || !d.isParent(id, parent)) {
```
Comment (mtrmac) [inline drivers/overlay/overlay.go:2165]: A random drive-by: Is it really correct to drop the special handling of `!d.isParent`?

Comment (vrothberg) [reply drivers/overlay/overlay.go:2165]: Thanks!

I think it doesn't matter for the current callers but we probably should do a naive diff if it's not the parent.
@giuseppe WDYT?

Comment (giuseppe) [reply drivers/overlay/overlay.go:2165]: yes, I also think we should do the naive diff in this case.

Thanks @mtrmac for spotting it
